python/lang/correctness/pytest-assert_match-after-path-patch.yaml

```diff
@@ -0,0 +1,23 @@
+rules:
+  - id: pytest-assert_match-after-path-patch
+    patterns:
+      - pattern-inside: |
+          import pytest
+          ...
+      - pattern-either:
+        - pattern-inside: |
+            mocker.patch("pathlib.Path", $MOCKED_VALUE)
+            ...
+        - pattern-inside: |
+            mocker.patch.object(Path, $METHOD, $MOCKED_VALUE)
+            ...
+      - pattern:
+          snapshot.assert_match(...)
+    message: >-
+      snapshot.assert_match makes use of pathlib to create files. Patching $METHOD may result in unexpected snapshot behavior
```

Comment:
`$METHOD` may not be set here - are we okay with this behavior for community adoption?  cc @minusworld @raghavjain3